File: src/go/twosum/twosum.go
package twosum 

import (
  "fmt"
)

func twoSum(nums []int, target int) []int {
    m := make(map[int]int)
    ret := make([]int, 2)
    for i,j := range nums{
        if _, ok:=m[j] ; ok {
		ret[0], ret[1] = m[j]+1, i+1
		break
	}else{
		m[target-j] = i
	}
    }
    return ret
}

func main(){
    nums:=[]int{2,7,9,11}
    target:=9
    fmt.Println(twoSum(nums, target))
}

File: src/java/Twosum.java
import java.util.List;
import java.util.Map;
import java.util.HashMap;

public class Twosum {
    public int[] twoSum(int[] numbers, int target) {
        Map<Integer, Integer> map = new HashMap<Integer, Integer>();
		for (int i=0 ; i<numbers.length; i++ ){
			if ( map.containsKey(numbers[i]) ){
				return new int[]{map.get(numbers[i])+1, i+1};
			}else{
				map.put(target-numbers[i], i);
			}
		}
		return null;
    }

    public static void main(String argvi[]){
	Twosum t = new Twosum();
 	int[] nums = {2,7,9,11};
	int target = 9;
	int[] ret = t.twoSum(nums, target);
	for (int i : ret)
		System.out.println(i);
    }
}
